corehq/apps/reports/standard/sms.py

```diff
@@ -260,6 +280,9 @@ def rows(self):
             if not message_type_filter(message_types):
                 continue
 
+            if reporting_locations_id and message.location_id not in reporting_locations_id:
```

Comment:
This lookup can still perform slowly when there are many location ids in reporting_locations_id because in order to check if a location_id is in this list it has to iterate through all elements in the list in a worse case scenario.  I won't block on it, but we should make it so that reporting_locations_id is a dictionary or set so that the lookup happens much faster (see http://stackoverflow.com/questions/513882/python-list-vs-dict-for-look-up-table)